src/libraries/System.Private.CoreLib/src/System/ComponentModel/DefaultValueAttribute.cs

```diff
@@ -35,6 +39,12 @@ public class DefaultValueAttribute : Attribute
             // The null check and try/catch here are because attributes should never throw exceptions.
             // We would fail to load an otherwise normal class.
 
+            if (!IsSupported)
+            {
+                Debug.Assert(!IsSupported, "Runtime instantiation of this attribute is not allowed.");
```

Comment:
Assert vs. throw approaches - was the Assert approach taken to be backwards compatible here?

Reply:
Yes, respecting the comments on the type so as not to throw